src/main/java/net/neoforged/neoforge/event/entity/living/SlimeSplitEvent.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.entity.living;
+
+import java.util.List;
+import net.minecraft.world.entity.Entity.RemovalReason;
+import net.minecraft.world.entity.monster.Slime;
+import net.neoforged.bus.api.Event;
+import net.neoforged.bus.api.ICancellableEvent;
+
+/**
+ * This event is fired whenever a slime is removed and splits into multiple child slimes.
```

Comment:
I think it's worth adding that this fires for magma cubes and potentially other subclasses of Slime too.